### DISCUSSION TITLE
Spring modulith test and small problem with ApplicationModules test when using HATEOS

### DISCUSSION THREAD

**Discussion by bloowper** [spring-projects/spring-modulith#487] [Q&A]
Hello Everyone,

I am currently working on a small project  for fun using Spring and modulith, and I've encountered an issue where I need some guidance. My challenge involves disabling REST controllers in the adapter layer during ApplicationModules testing. I aim to leverage Spring HATEOAS and the linkTo(methodOn(XController.class)) function, but the presence of these dependencies in my tests leads to failures.

I am looking for a solution that aligns with the spirit of modulith yet allows for a workaround in this particular scenario. This would facilitate easier resource linking management. Has anyone faced a similar challenge or knows of an elegant way to disable these controllers in tests without compromising clean architecture principles?

Any suggestions or advice would be immensely appreciated. Thank you in advance for your help and insights.

Best regards,
Tomasz Orchowski

example controller:
``` java
package orchowski.tomasz.carrental.fleetinventory.adapter;

@RestController
@RequestMapping("/vehicles")
@RequiredArgsConstructor
public class FleetInventoryRestAdapter {
    private final FleetInventoryManagerFacade vehicleManagementFacade;

    @PostMapping
    EntityModel<VehicleDto> addNewVehicleToFleet(@RequestBody AddNewVehicleCommand addNewVehicleCommand){
        var vehicle = vehicleManagementFacade.addNewVehicleToFleet(addNewVehicleCommand);
        return createEntityModel(vehicle);
    }

    @GetMapping("/{id}")
    EntityModel<VehicleDto> getVehicle(@PathVariable UUID id){
        var vehicle = vehicleManagementFacade.getVehicle(id);
        return createEntityModel(vehicle);
    }

    private EntityModel<VehicleDto> createEntityModel(VehicleDto vehicle) {
        EntityModel<VehicleDto> vehicleModel = EntityModel.of(vehicle);
        vehicleModel.add(
                linkTo(methodOn(FleetInventoryRestAdapter.class)
                        .getVehicle(vehicle.id()))
                        .withSelfRel()
                        .withType("GET")
                        .withTitle("Vehicle details"),
                linkTo(methodOn(FleetMaintenanceRest.class)
                        .getServiceUnit(vehicle.id()))
                        .withRel("service-unit")
                        .withType("GET")
                        .withTitle("Service unit details")
        );
        return vehicleModel;
    }

}

```

**Comment by odrotbohm** [comment]
It's a bit unclear what you're asking.

> the presence of these dependencies in my tests leads to failures.

What does that mean?

> Has anyone faced a similar challenge or knows of an elegant way to disable these controllers in tests without compromising clean architecture principles?

Which challenges? Which controllers? From looking at a single controller, it's impossible to guess what could be wrong.
